relations/jf/src/note.rs

```diff
@@ -102,16 +110,19 @@ impl NoteGadget for PlonkCircuit<CircuitField> {
 impl PublicInput for SourcedNote {
```

Comment:
I'm not sure why there even exists an `impl PublicInput for SourcedNote`: my understanding was that public/private/full inputs are for _relations_ and `SourcedNote` is a value in that relation; relation == function, note == input to the relation. I understand that it might be easier to write it that way and require at the type-level that `forall A exists A: PublicInput` but from a "domain" perspective (ie in SNARK theory?) there is no public input for a note, correct?